### PR TITLE
add top 10 beneficiaries chart - part 1

### PR DESCRIPTION
**Pull request by jerem1508**

### CODE REVIEW

File: client/src/pages/european-projects/charts-config.json [hunk 1]
```diff
@@ -41,5 +41,14 @@
     "source": "Commission européenne, Cordis",
     "sourceURL": "https://cordis.europa.eu/",
     "integrationURL": "/european-projects/components/pages/analysis/overview/charts/main-beneficiaries"
+  },
+  {
+    "id": "top10benefifiaries",
```
Comment (annelhote) [inline client/src/pages/european-projects/charts-config.json:46]: top10beneficiaries

File: client/src/pages/european-projects/components/pages/analysis/positioning/charts/top-10-beneficiaries/index.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  const { data, isLoading } = useQuery({
+    queryKey: ["Top10Beneficiaries", params],
+    queryFn: () => GetData(params)
+  })
+
+  if (isLoading || !data) return <Template />
+  return (
+    <ChartWrapper
+      id="top10benefifiaries"
```
Comment (annelhote) [inline client/src/pages/european-projects/components/pages/analysis/positioning/charts/top-10-beneficiaries/index.tsx:23]: top10beneficiaries

File: client/src/pages/european-projects/components/pages/analysis/positioning/charts/top-10-beneficiaries/options.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        autoRotation: [-45, -90],
+        style: {
+          fontSize: '13px',
+          fontFamily: 'Verdana, sans-serif'
+        }
+      }
+    },
+    yAxis: [
+      {
+        // min: 0,
```
Comment (annelhote) [inline client/src/pages/european-projects/components/pages/analysis/positioning/charts/top-10-beneficiaries/options.tsx:24]: delete commented code

File: client/src/pages/european-projects/components/pages/analysis/positioning/charts/top-10-beneficiaries/query.tsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+const { VITE_APP_SERVER_URL } = import.meta.env;
+
+export async function GetData(params: string) {
+  let url = `${VITE_APP_SERVER_URL}/european-projects/analysis-positioning-top-10-beneficiaries`;
+  if (params !== '') {
+    url = `${VITE_APP_SERVER_URL}/european-projects/analysis-positioning-top-10-beneficiaries?${params}`;
+  }
+
+  return fetch(url).then((response) => (response.json()))
```
Comment (annelhote) [inline client/src/pages/european-projects/components/pages/analysis/positioning/charts/top-10-beneficiaries/query.tsx:9]: return fetch(url).then((response) => response.json());

no parentheses needed ?

File: client/src/pages/european-projects/components/pages/analysis/positioning/charts/top-10-beneficiaries/query.tsx [hunk 5]
```diff
@@ -0,0 +1,11 @@
+const { VITE_APP_SERVER_URL } = import.meta.env;
+
+export async function GetData(params: string) {
+  let url = `${VITE_APP_SERVER_URL}/european-projects/analysis-positioning-top-10-beneficiaries`;
```
Comment (annelhote) [inline client/src/pages/european-projects/components/pages/analysis/positioning/charts/top-10-beneficiaries/query.tsx:4]: const url = `${VITE_APP_SERVER_URL}/european-projects/analysis-positioning-top-10-beneficiaries${params ? `?${params}` : ''}`;

simplier ?

File: client/src/pages/european-projects/components/pages/analysis/positioning/charts/top-10-beneficiaries/utils.tsx [hunk 6]
```diff
@@ -0,0 +1,5 @@
+export function getDefaultParams(searchParams) {
+  const params = [...searchParams].map(([key, value]) => `${key}=${value}`).join('&');
+
+  return params + '&stage=successful';
```
Comment (annelhote) [inline client/src/pages/european-projects/components/pages/analysis/positioning/charts/top-10-beneficiaries/utils.tsx:4]: I would have written :
params.append('stage', 'successful');
return params.toString()

Comment (annelhote) [inline client/src/pages/european-projects/components/pages/analysis/positioning/charts/top-10-beneficiaries/utils.tsx:4]: You might want to check if the "stage" param is not already set ?

File: client/src/pages/european-projects/components/pages/analysis/positioning/index.tsx [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import { Container } from "@dataesr/dsfr-plus";
+import { useSearchParams } from "react-router-dom";
+import { useEffect } from "react";
+import Top10Beneficiaries from "./charts/top-10-beneficiaries";
+
+export default function Positioning() {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  useEffect(() => {
+    if (searchParams.get('country_code') === null) {
```
Comment (annelhote) [inline client/src/pages/european-projects/components/pages/analysis/positioning/index.tsx:12]: if (!searchParams.has('country_code')) {

File: client/src/pages/european-projects/components/pages/analysis/positioning/index.tsx [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+      setSearchParams({ country_code: 'FRA' });
```
Comment (annelhote) [inline client/src/pages/european-projects/components/pages/analysis/positioning/index.tsx:13]: searchParams.append('country_code': 'FRA');
setSearchParams(searchParams);

I do not know if you need to do it through a copy of searchParams ?

**Review by annelhote — CHANGES_REQUESTED** [review]

File: client/src/pages/european-projects/components/pages/analysis/positioning/charts/top-10-beneficiaries/utils.tsx (hunk 6 above)
Comment (jerem1508) [reply client/src/pages/european-projects/components/pages/analysis/positioning/charts/top-10-beneficiaries/utils.tsx:4]: cleaner

**Review by annelhote — APPROVED** [review]

**Pull request merged**